### PR TITLE
OHRM5X-1699: Handle dashboard - user role permissions

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

File: installer/Migration/V5_2_0/permission/data_group.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+dashboard_admin_widgets:
```
Comment (RajithaKumara) [inline installer/Migration/V5_2_0/permission/data_group.yaml:1]: better to split `subunit`, `location`
`dashboard_subunit_widgets`
`dashboard_location_widgets`

File: installer/Migration/V5_2_0/permission/data_group.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+dashboard_configuration:
+  description: 'Dashboard - Configuration'
+  allowed:
+    read: true
+    create: false
+    update: true
+    delete: false
+  permissions:
+    - { role: Admin, permission: { read: true, create: false, update: true, delete: false, self: false } }
```
Comment (RajithaKumara) [inline installer/Migration/V5_2_0/permission/data_group.yaml:41]: what about other widgets,
Action summary. all together 6 widgets right

File: src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php [hunk 3]
```diff
@@ -51,8 +53,20 @@ public function updateModuleStatus(array $modules): array
...
+                    $this->updateDataGroupPermissionForWidgetModules(
+                        'dashboard_leave_widget',
+                        (bool)$modules[$module->getName()]
+                    );
+                } elseif ($module->getName() === 'time') {
+                    $this->updateDataGroupPermissionForWidgetModules(
+                        'dashboard_time_widget',
+                        (bool)$modules[$module->getName()]
+                    );
+                }
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php:69]: better to move this to service layer or API layer

File: src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php [hunk 4]
```diff
@@ -70,4 +84,51 @@ public function getDisabledModuleList(): array
         $q->select('m.name');
         return $q->getQuery()->execute();
     }
+
+    /**
+     * @return Module[]
+     */
+    public function getEnabledModuleList(): array
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php:91]:
```suggestion
    public function getEnabledModuleNameList(): array
```

File: src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php [hunk 5]
```diff
@@ -70,4 +84,51 @@ public function getDisabledModuleList(): array
         $q->select('m.name');
         return $q->getQuery()->execute();
     }
+
+    /**
+     * @return Module[]
```
Comment (RajithaKumara) [inline src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php:89]:
```suggestion
     * @return string[]
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php (hunk 3 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php:69]: Added as event dispatcher

File: src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php (hunk 5 above)
Comment (SandeepaOHRM) [reply src/plugins/orangehrmCorePlugin/Dao/ModuleDao.php:89]: Fixed

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**